pandaserver/dataservice/adder_gen.py

```diff
@@ -460,74 +479,80 @@ def run(self):
                                     forAnal=True,
                                 )[0]
                                 if self.job is None:
-                                    self.logger.debug(f": associated job PandaID={assJobID} not found in DB")
+                                    self.logger.debug(f": associated job PandaID={associate_job_id} not found in DB")
                                 else:
-                                    cThr = closer.Closer(self.taskBuffer, assDBlocks, assJob)
-                                    self.logger.debug(f"start Closer for PandaID={assJobID}")
-                                    # cThr.start()
-                                    # cThr.join()
-                                    cThr.run()
-                                    del cThr
-                                    self.logger.debug(f"end Closer for PandaID={assJobID}")
+                                    closer_thread = closer.Closer(self.taskBuffer, associate_dispatch_blocks, associate_job)
+                                    self.logger.debug(f"start Closer for PandaID={associate_job_id}")
+                                    closer_thread.run()
+                                    del closer_thread
+                                    self.logger.debug(f"end Closer for PandaID={associate_job_id}")
             duration = datetime.datetime.now(datetime.timezone.utc).replace(tzinfo=None) - start_time
             self.logger.debug("end: took %s.%03d sec in total" % (duration.seconds, duration.microseconds / 1000))
-            # try:
-            #     # remove Catalog
-            #     os.remove(self.xmlFile)
-            # except Exception:
-            #     pass
+
             # remove Catalog
-            self.taskBuffer.deleteJobOutputReport(panda_id=self.jobID, attempt_nr=self.attemptNr)
+            self.taskBuffer.deleteJobOutputReport(panda_id=self.job_id, attempt_nr=self.attempt_nr)
             del self.data
             del report_dict
         except Exception as e:
-            errStr = f": {str(e)} {traceback.format_exc()}"
-            self.logger.error(errStr)
+            err_str = f": {str(e)} {traceback.format_exc()}"
+            self.logger.error(err_str)
             duration = datetime.datetime.now(datetime.timezone.utc).replace(tzinfo=None) - start_time
             self.logger.error("except: took %s.%03d sec in total" % (duration.seconds, duration.microseconds / 1000))
             # unlock job output report
             self.taskBuffer.unlockJobOutputReport(
-                panda_id=self.jobID,
-                attempt_nr=self.attemptNr,
+                panda_id=self.job_id,
+                attempt_nr=self.attempt_nr,
                 pid=self.pid,
                 lock_offset=self.lock_offset,
             )
 
     # parse XML
     # 0: succeeded, 1: harmless error to exit, 2: fatal error, 3: event service
-    def parseXML(self):
+    def parse_xml(self):
```

Comment:
Check atlas_panda.Job_Output_Report. Reports seem to be in json already, but it would be good to talk with Paul/Tadashi as they know the history and can confirm. Ideally we would remove XML code.

Reply:
This development also affects positively the dispatcher and the copyArchive. They have variables with XML names, although it's json already.